### ISSUE TITLE
Update release system to release multiple groups of packages in a single step

### ISSUE THREAD

**Issue by heliocliu** [microsoft/FluidFramework#8056]
Current process for release is that it's run for each layer that needs releasing, then the next dependent layer updates to use the just released dependency, and so on.  Once independent versions are supported in the same repo, the release system should be able to do the whole release in a single user-initiated action instead of one for each version layer.

e.g. if client and server merge into the same repo, the system should determine that server is at the bottom of the dependencies, run a release on it, update client to depend on the new server versions, ensure the just release packages are available, then release the new client packages.

This item is dependent on the independent versioning/breaking change detection work to support the existing package layers in the same repo, but does not need any further subdivision of packages.  It's not dependent on #8054 and #8055 for the new version tracking process, but will probably need updates to accommodate those changes
